src/switch_rtp.c

```diff
@@ -7633,28 +7633,37 @@ static switch_status_t process_rtcp_xr_report(switch_rtcp_report_data_t *report_
 					}
 					report_data->rtcp_data.xr_blocks[count].xr_pt = header->bt;
 					report_data->rtcp_data.xr_blocks[count].report_data.rr_time = rr_time;
-					}
-					break;
-				case XR_VOIP_METRICS: {
+					++count;
+				}
+				break;
+			case XR_VOIP_METRICS:
+				{
 					switch_rtcp_xr_rb_voip_metrics *voip_metrics = (switch_rtcp_xr_rb_voip_metrics *) header;
 					switch_log_printf(SWITCH_CHANNEL_SESSION_LOG(rtp_session->session), SWITCH_LOG_DEBUG,
 						"RTCP XR VoIP metrics: loss rate: %02x rt delay: %d es delay: %d \n", voip_metrics->loss_rate, voip_metrics->round_trip_delay, voip_metrics->end_system_delay);
 					report_data->rtcp_data.xr_blocks[count].xr_pt = header->bt;
 					report_data->rtcp_data.xr_blocks[count].report_data.voip_metrics = voip_metrics;
+					++count;
 				}
 				break;
-				case XR_STATS: {
+			case XR_STATS:
+				{
 					switch_rtcp_xr_rb_stats *stats = (switch_rtcp_xr_rb_stats *) header;
+					switch_log_printf(SWITCH_CHANNEL_SESSION_LOG(rtp_session->session), SWITCH_LOG_DEBUG,
+						"RTCP XR Statistics summary: begin seq %d, end seq %d, lost packets %d\n", stats->begin_seq, stats->end_seq, stats->lost_packets);
 					report_data->rtcp_data.xr_blocks[count].xr_pt = header->bt;
 					report_data->rtcp_data.xr_blocks[count].report_data.stats = stats;
+					++count;
 				}
-				default:
-					break;
+			default:
+				break;
 			}
-			++count;
 		}
 		header = (switch_rtcp_xr_rb_header *)((int32_t *)header + blen + 1);
 	}
+	report_data->xr_count = count;
+	switch_log_printf(SWITCH_CHANNEL_SESSION_LOG(rtp_session->session), SWITCH_LOG_DEBUG, "Parsed %d report blocks\n", count);
```

Comment:
I would suggest to remove or increment it to DEBUG1..9 if you need this log. I think when we start processing XR reports this will pollute our cli logs or b2bua logs. Same thing for line 7902, 7581, 7606

Reply:
I left the one in 7902, since that warning was already there